circle.yml

```diff
@@ -1,38 +1,109 @@
-        "$CIRCLE_SHA1"
-        "$CIRCLE_TAG"
-        "$CIRCLE_PROJECT_USERNAME"
-        "$CIRCLE_PROJECT_REPONAME"
-        "$CIRCLE_BUILD_URL"
-        > version.json
-    - docker build -t app:build -f Dockerfile.deploy .
-
-test:
-  override:
-    - echo "UI Tests are run on Travis CI. https://travis-ci.org/mozilla/addons-server/"
-
-deployment:
-  latest:
-    branch: master
-    commands:
-      - docker tag app:build ${DOCKERHUB_REPO}:latest
-      - docker login -e $DOCKERHUB_EMAIL -u $DOCKERHUB_USER -p $DOCKERHUB_PASS
-      - docker push ${DOCKERHUB_REPO}:latest
-
-  releases:
-    # push all tags
-    tag: /.*/
-    commands:
-      - docker tag app:build ${DOCKERHUB_REPO}:${CIRCLE_TAG}
-      - docker login -e $DOCKERHUB_EMAIL -u $DOCKERHUB_USER -p $DOCKERHUB_PASS
-      - docker push ${DOCKERHUB_REPO}:${CIRCLE_TAG}
+jobs:
+  build:
+    machine: true
+    working_directory: ~/addons-server
+    steps:
+      - checkout
+      - run: >
+          printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n'
+          "$CIRCLE_SHA1"
+          "$CIRCLE_TAG"
+          "$CIRCLE_PROJECT_USERNAME"
+          "$CIRCLE_PROJECT_REPONAME"
+          "$CIRCLE_BUILD_URL"
+          > version.json
+      - run: docker build -t app:build -f Dockerfile.deploy .
+      - persist_to_workspace:
+          root: .
+          paths: .
+  integration_test:
+    working_directory: ~/addons-server
+    machine: true
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - uitest-cache-{{ checksum "requirements/docs.txt" }}
+            - uitest-cache-{{ checksum "requirements/prod.txt" }}
+            - uitest-cache-{{ checksum "requirements/dev.txt" }}
+            - uitest-cache-{{ checksum "tests/ui/requirements.txt" }}
+            - uitest-cache-
+      - run:
+          name: Install Docker Compose
+          command: |
+            set -x
+            sudo pip install docker-compose
+      - run:
+          name: Start container, verify it's running and start tests
+          command: |
+            set -x
+            sudo sysctl -w vm.max_map_count=262144
+            docker-compose up -d
+            sleep 10
+            docker-compose exec web bash /code/scripts/ui-test.sh
+      - store_artifacts:
+          path: ui-test-results
+      - save_cache:
+          key: uitest-cache-{{ checksum "requirements/docs.txt" }}
+          paths:
+            - .tox
+      - save_cache:
+          key: uitest-cache-{{ checksum "requirements/prod.txt" }}
+          paths:
+            - .tox
+      - save_cache:
+          key: uitest-cache-{{ checksum "requirements/dev.txt" }}
+          paths:
+            - .tox
+      - save_cache:
+          key: uitest-cache-{{ checksum "tests/ui/requirements.txt" }}
+          paths:
+            - .tox
+      - save_cache:
+          key: uitest-cache-
+          paths:
+            - .tox
```

Comment:
Guess that's one key too much again heh

Reply:
https://circleci.com/docs/2.0/caching/

Reply:
Same here, the key does not reference any file 😕